typhos/notes.py

```diff
@@ -0,0 +1,189 @@
+import logging
+import os
+from datetime import datetime
+from enum import IntEnum
+from pathlib import Path
+from typing import Dict, Optional, Tuple
+
+import platformdirs
+import yaml
+from qtpy import QtWidgets
+
+from typhos import utils
+
+logger = logging.getLogger(__name__)
+NOTES_VAR = "PCDS_DEVICE_NOTES"
+
+
+class NotesSource(IntEnum):
+    USER = 0
+    ENV = 1
+    # HAPPI = 2  # to be implemented later
+
+
+def get_data_from_yaml(device_name: str, path: Path) -> Optional[Dict[str, str]]:
+    """
+    Returns the device data from the yaml file stored in ``path``.
+    Returns `None` if reading the file fails or there is no information
+
+    Parameters
+    ----------
+    device_name : str
+        The name of the device to retrieve notes for.
+    path : Path
+        Path to the device information yaml
+
+    Returns
+    -------
+    Optional[Dict[str, str]]
+        The device information or None
+    """
+    try:
+        with open(path) as f:
+            device_notes = yaml.full_load(f)
+    except Exception as ex:
+        logger.warning(f'failed to load device notes: {ex}')
+        return
+
+    if device_name in device_notes:
+        return device_notes.get(device_name, None)
```

Comment:
The if statement is redundant with the `.get` here
```suggestion
    return device_notes.get(device_name, None)
```